Automattic/jetpack
# Lazy Images: Add proper placeholders to minimize layout shift

Partial retry of #10086. Also addresses #8995, which has since been closed by a user.

#### Changes proposed in this Pull Request:

* From a user's perspective, there should be much less jank, or layout shift as lazy images are loaded.
* From an implementation perspective, we are loading the placeholder in the `src` tag instead of `srcset`. We originally moved to `srcset` in #10054, but `srcset` doesn't support using an SVG, and I don't think using `srcset` is necessary. Since we'll be loading proper placeholder, I believe that the originally reported issue in #8879 should still be addressed.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

* This is a fix for a longstanding issue where lazily loaded images caused lots of layout shift.

#### Does this pull request change what data or activity we track or use?

No data changes.

#### Testing instructions:

* Check out pull request
* In a post/page, add some combination of content that includes images
* Load the page
* Observe that there are blank spaces where the image would load
* Observe that the image eventually loads and replaces the blank space

**After**


![eric_blog_after](https://user-images.githubusercontent.com/1126811/83483848-3707c980-a469-11ea-861b-28841679a874.gif)

**Before**

![eric_blog_before](https://user-images.githubusercontent.com/1126811/83483892-41c25e80-a469-11ea-9436-34b93a6db887.gif)

#### Proposed changelog entry for your changes:

* Minimizes layout shift caused by lazily loaded images by using placeholders with a correct aspect ratio.
